noxfile.py

```diff
@@ -261,23 +204,40 @@ def run_tests_with_dev_version_of(session: nox.Session, repository: str) -> None
             "https://pypi.anaconda.org/scientific-python-nightly-wheels/simple",
             "numpy",
         )
+    else:
+        session.install(f"git+{repository}")
+
     session.install(".[tests]")
     session.run(*pytest_command, *session.posargs)
 
 
-sphinx_commands: tuple[str, ...] = (
+doc_build_dir = "$READTHEDOCS_OUTPUT" if running_on_rtd else "docs/build/html"
+
+if running_on_rtd:
+    rtd_output_path = pathlib.Path(os.environ.get("READTHEDOCS_OUTPUT")) / "html"
+    rtd_output_path.mkdir(parents=True, exist_ok=True)
+    doc_build_dir = str(rtd_output_path)
+else:
+    doc_build_dir = "docs/build/html"
+
+sphinx_base_command: list[str] = [
     "sphinx-build",
     "docs/",
-    "docs/build/html",
+    doc_build_dir,
     "--nitpicky",
-    "--fail-on-warning",
     "--keep-going",
-    "-q",
-)
+]
+
+if not running_on_rtd:
+    sphinx_base_command.extend(
+        [
+            "--fail-on-warning",
+            "--quiet",
```

Comment:
Usually, the documentation output is much easier to parse with the `--quiet` option. However, doing the RTD build without the `--quiet` option will let us see a more complete log of the documentation build without having to do a repeat run.  Might be helpful to add this tip to the documentation guide.